fms/modules/attention.py

```diff
@@ -262,32 +262,35 @@ def __init__(
         assert torch.distributed.is_initialized()
 
         rank, world_size = distributed.rank_and_world(group)
-        assert (
-            nheads % world_size == 0
-        ), "The number of heads must be divisible by world size"
+        assert nheads % world_size == 0 and (
+            kvheads % world_size == 0 or world_size % kvheads == 0
+        ), "The number of heads must be divisible by world size or the world size must be a multiple of kv heads"
         MultiHeadAttention.__init__(
             self,
             emb_dim,
             emb_kq,
             emb_v,
             nheads // world_size,
-            (kvheads // world_size) if kvheads > 1 else kvheads,
+            (kvheads // world_size) if kvheads >= world_size else 1,
```

Comment:
is this condition compatible with the `world_size % kvheads` assertion from above?

we allow world_size divisible by kvheads, a case which would imply kv heads smaller than world_size, so I guess this is ensuring we get a 1 instead of 0. Maybe a comment would help clarify the goals here and what our final distribution is supposed to look like relative to different configurations.

Reply:
I just mentioned the assert is wrong, so if it gets fixed it is compatible